### PR TITLE
Partially unrevert "Use nogo for code analysis instead of go vet"

### PR DESCRIPTION
**Pull request by Katharine**
Like #11602, but without actually removing the job because that makes it impossible to merge anything.

/cc @fejta

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 1]
```diff
@@ -221,7 +221,6 @@ presubmits:
   - name: pull-test-infra-verify-govet
     branches:
     - master
-    always_run: true
```
Comment (fejta) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: leave this and `optional: true`?

Comment (Katharine) [reply config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: done

**Comment by Katharine** [comment]
/retest
:roll_eyes:

**Pull request merged**